### PR TITLE
[Core] Adding ResetFlag to VariableUtils

### PR DESCRIPTION
**Pull request by loumalouomega**
Simple as that. Adding test

### CODE REVIEW

File: kratos/utilities/variable_utils.h [hunk 1]
```diff
@@ -450,16 +450,42 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
...
+    void ResetFlag(
+        const Flags& rFlag,
+        TContainerType& rContainer
+        )
+    {
+        KRATOS_TRY
+
+        const auto it_cont_begin = rContainer.begin();
+
+        #pragma omp parallel for
```
Comment (rubenzorrilla) [inline kratos/utilities/variable_utils.h:480]:
```suggestion
        #pragma omp parallel for firstprivate(it_cont_begin)
```

Comment (loumalouomega) [reply kratos/utilities/variable_utils.h:480]: It is not needed when it is const, we already had this discussion in a previous PR

File: kratos/utilities/variable_utils.h [hunk 2]
```diff
@@ -473,9 +499,11 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
     {
         KRATOS_TRY
 
+        const auto it_cont_begin = rContainer.begin();
+
         #pragma omp parallel for
```
Comment (rubenzorrilla) [inline kratos/utilities/variable_utils.h:504]:
```suggestion
        #pragma omp parallel for firstprivate(it_cont_begin)
```

Comment (loumalouomega) [reply kratos/utilities/variable_utils.h:504]: Idem

**Review by rubenzorrilla — COMMENTED** [review]
Just a minor comment on the OpenMP pragma.

**Comment by loumalouomega** [comment]
> Just a minor comment on the OpenMP pragma.

Not needed, constant values are threadsafe

**Review by rubenzorrilla — APPROVED** [review]

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**